nvflare/app_opt/pt/fedopt.py

```diff
@@ -215,12 +220,28 @@ def shareable_to_learnable(self, shareable: Shareable, fl_ctx: FLContext) -> Lea
             weights[key] = weights[key].detach().cpu().numpy()
         secs_detach = time.time() - start
 
+        # update unnamed parameters such as batch norm layers if there are any using the averaged update
+        base_model = fl_ctx.get_prop(AppConstants.GLOBAL_MODEL)
+        if not base_model:
+            self.system_panic(reason="No global base model!", fl_ctx=fl_ctx)
+            return base_model
+
+        base_model_weights = base_model[ModelLearnableKey.WEIGHTS]
+
+        n_fedavg = 0
+        for key, value in model_diff.items():
+            if key not in updated_params:
+                weights[key] = base_model_weights[key] + value
```

Comment:
Hi Holger, I want to make sure I understand this line correctly. This line will convert the model diff (unnamed parameters) back to the model avg since you did the (server weight - aggregated model diff).

Reply:
It basically does the standard FedAvg update for unnamed model parameters (i.e. batch norm stats). `value` here is the aggregated model differences.